osf/migrations/0177_update_schema_data.py

```diff
@@ -0,0 +1,282 @@
+
+def update_schema_configs(state, schema):
+    RegistrationSchema = state.get_model('osf', 'registrationschema')
+    for rs in RegistrationSchema.objects.all():
+        if rs.schema.get('description', False):
+            rs.description = rs.schema['description']
+            rs.save()
+
+def unset_schema_configs(state, schema):
+    RegistrationSchema = state.get_model('osf', 'registrationschema')
+    RegistrationSchema.objects.update(config=dict())
+
+def unmap_formblocks(state, schema):
+    RegistrationFormBlock = state.get_model('osf', 'registrationformblock')
+    RegistrationFormBlock.objects.all().delete()
+
+def noop(*args, **kwargs):
+    pass
+
+def create_block(state, schema_id, block_type, display_text='', required=False, help_text='',
+        answer_id='', chunk_id='', example_text=''):
+    RegistrationFormBlock = state.get_model('osf', 'registrationformblock')
+
+    return RegistrationFormBlock.objects.create(
+        schema_id=schema_id,
+        block_type=block_type,
+        required=required,
+        display_text=display_text,
+        help_text=help_text,
+        answer_id=answer_id,
+        chunk_id=chunk_id,
+        example_text=example_text
+    )
+
+# Split question multiple choice options into their own blocks
+def split_options_into_blocks(state, rs, question, chunk_id):
+    for option in question.get('options', []):
+        answer_text = option if isinstance(option, basestring) else option.get('text')
+        help_text = '' if isinstance(option, basestring) else option.get('tooltip', '')
+
+        create_block(
+            state,
+            rs.id,
+            'select-input-option',
+            display_text=answer_text,
+            help_text=help_text,
+            chunk_id=chunk_id,
+        )
+
+def get_answer_id(question):
+    """
+    Answer ids will map to the user's response
+    """
+    return question.get('qid', '') or question.get('id', '')
+
+def strip_html(string_with_html):
+    """
+    Some original schemas have html in the help text.  Just replacing this with empty strings.
+    """
+    stripped_html = re.sub('<.*?>', '', string_with_html)
+    return stripped_html
+
+def find_title_description_help_example(rs, question):
+    """
+    Schemas are inconsistent with regards to the information going into "title",
+    "description", and "help" blocks.
+
+    :returns tuple, title, description, help, example strings
+
+    """
+    title = question.get('title', '')
+    description = question.get('description', '')
+    help = strip_html(question.get('help', ''))
+    example = ''
+
+    schema_name = rs.schema.get('name', '')
+    # Descriptions that contain any of these keywords
+    # are turned into help text instead.
+    help_text_keywords = [
+        'please',
+        'choose',
+        'provide',
+        'format',
+        'describe',
+        'who',
+        'what',
+        'when',
+        'where',
+        'use',
+        'you',
+        'your',
+        'skip']
```

Comment:
Subjective list, but this seems to do a pretty good job of mapping descriptions to help_text instead.